source/extensions/stat_sinks/hystrix/hystrix.cc

```diff
@@ -304,6 +320,33 @@ void HystrixSink::flush(Stats::Source&) {
   incCounter();
   std::stringstream ss;
   Upstream::ClusterManager::ClusterInfoMap clusters = server_.clusterManager().clusters();
+
+  // Save a map of the relevant histograms per cluster in a convenient format.
+  std::unordered_map<absl::string_view, std::unordered_map<std::string, double>, StringViewHash>
+      time_histograms;
+  for (const Stats::ParentHistogramSharedPtr histogram : server_.stats().histograms()) {
```

Comment:
nit: I think we should be using the `Stats::Source::cachedHistograms()` method on the `Source` that's being passed in (https://github.com/envoyproxy/envoy/blob/346059548e135199eb0b7f0006f3ef19e173bf79/include/envoy/stats/stats.h#L301) to get the histograms.